design/thin-edge-core.md

```diff
@@ -4,24 +4,31 @@ A typical IoT application has to address:
 
 * __sophisticated business-cases__, with applications involving several vendors, actors and users;
 * __very diverse integration needs__ over operating systems, protocols, data sources, processing tools, actuators, cloud end-points;
-* __constraint environments__ with low-resources devices, large-scale deployments, high exposure to security risks,
-  prohibitive manual operations, long-term supports;
-* __safety related requirements__ with devices operating machines in hazardous contexts.
+* __constraint environments__ with low-resources devices, high exposure to security risks, prohibitive manual operations;
+* __particular non-functional requirements__ safety of devices operating machines in hazardous contexts,
+  scalability over large fleet of devices, long-term supports of scattered devices.
   
 The forces behind these requirements are pushing in opposite directions.
 
-* On one side, from the use-case and integration perspective, one needs a dynamic and open setting with a large set of features that can be easily extended and combined.
+* On one side, to address the diversity of use-cases and integration requirements, one needs a large open set of features that can be easily extended and combined.
   [NodeRed](https://nodered.org/) is the archetype of such tools facilitating connections of components.
 * On the other side, to meet resource and security constraints, one needs a minimal system specifically built for a task with cherry-picked components.
-  On the extreme side this leads to the idea of [unikernel](https://en.wikipedia.org/wiki/Unikernel) with sealed, single-purpose software images.
+  On the extreme side, this leads to the idea of [unikernel](https://en.wikipedia.org/wiki/Unikernel) with sealed, single-purpose software images.
   
-Can these two poles be reconciled?
+What makes thin-edge unique is its approach to reconcile these two poles with two levels of building blocks.
+* To maximize flexibility and interoperability, thin-edge provides the tools to build an agent from independent executables that interact over MQTT using JSON messages.
+* To minimize resources and vulnerabilities, thin-edge features fine-grain Rust components as well as the rules to combine them into purpose-specific agents.
+* To make feasible the assemblage of fine-grain and MQTT-based components into IIoT agents,
```

Comment:
To make feasible the assemblage of fine-grain Rust and MQTT-based components into IIoT agents,